plugins/follow_me/follow_me.h

```diff
@@ -0,0 +1,206 @@
+#pragma once
+
+#include <iostream>
+#include <cmath>
+
+
+namespace dronecore {
+
+class FollowMeImpl;
+
+class FollowMe
+{
+public:
+    explicit FollowMe(FollowMeImpl *impl);
+    ~FollowMe();
+
+    /**
+     * @brief Results of FollowMe operations
+     */
+    enum class Result {
+        SUCCESS = 0, /**< @brief %Request succeeded. */
+        NO_DEVICE, /**< @brief No device connected. */
+        CONNECTION_ERROR, /**< @brief %Connection error. */
+        BUSY, /**< @brief Vehicle busy. */
+        COMMAND_DENIED, /**< @brief Command denied. */
+        TIMEOUT, /**< @brief %Request timeout. */
+        UNKNOWN /**< @brief Unknown error. */
+    };
+
+    /**
+     * @brief FollowMe configuration.
+     * Reference: https://docs.px4.io/en/advanced_config/parameter_reference.html#follow-target
+     */
+    class Configuration
+    {
+    public:
+        /**
+         * @brief Side to follow target from
+         */
+        enum class FollowDirection {
+            NONE = -1,
+            FRONT_RIGHT = 0, /**< @brief Follow from front right */
+            BEHIND = 1, /**< @brief Follow from behind */
+            FRONT = 2, /**< @brief Follow from front */
+            FRONT_LEFT = 3 /**< @brief Follow from front left */
+        };
+
+        // Default configuration
+        ///////////////////////////////////////////////////////////
+        static constexpr const float DEF_HEIGHT_ABOVE_HOME =
+            8.0f; /**< @brief Default follow target altitude */
+        static constexpr const float MIN_HEIGHT_ABOVE_HOME = 8.0f; /**< @brief Min follow target altitude */
+
+        static constexpr const float DEF_DIST_WRT_FT = 8.0f; /**< @brief Distance to follow target from */
+        static constexpr const float MIN_DIST_WRT_FT =
+            1.0f; /**< @brief Min distance to follow target from */
+
+        static constexpr const FollowDirection DEF_FOLLOW_DIR =
+            FollowDirection::BEHIND; /**< @brief Side to follow target from */
+        static constexpr const float DEF_DYN_FLT_ALG_RSP =
+            0.5f; /**< @brief Default Dynamic filtering algorithm responsiveness to target movement */
+        static constexpr const float MIN_DYN_FLT_ALG_RSP =
+            0.0f; /**< @brief Min Dynamic filtering algorithm responsiveness to target movement */
+        static constexpr const float MAX_DYN_FLT_ALG_RSP =
+            1.0f; /**< @brief Max Dynamic filtering algorithm responsiveness to target movement */
+        ///////////////////////////////////////////////////////////
+
+        // FollowMe Configuration Accessors
+        /**
+         * @brief min_height_m
```

Comment:
All these accessors need proper docs. For example, this one is "Minimum follow height, in metres. "